### PR TITLE
US4511 Add as Trip Participant

### PR DESCRIPTION
**Pull request by silbermm**
When a applicant submits a trip application, make them a participant on the trip. This means creating a donor account if they don't already have one, creating them a pledge on the correct pledge campaign and adding them as a participant on the correct groups and event.

### CODE REVIEW

File: CI/SQL/20160729_112800_US4511-StoredProcToGetInfoForGoApplicant.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	SET NOCOUNT ON;
+
+	DECLARE @GoTripEventTypeID INT = 6
+
+	SELECT pc.Destination_ID, pc.Fundraising_Goal, g.group_id
+	FROM [dbo].[Pledge_Campaigns] pc
+	JOIN dbo.events e ON e.program_id = pc.program_id AND e.Event_Type_ID = @GoTripEventTypeID
+	JOIN dbo.event_groups eg ON eg.event_id = e.event_id
+	JOIN groups g ON g.group_id = eg.group_id
+	WHERE pledge_campaign_id = 10000000
```
Comment (andycanterbury) [inline CI/SQL/20160729_112800_US4511-StoredProcToGetInfoForGoApplicant.sql]: This is the hard coded ID for GO Midgar. Guessing this should be @Pledge_Campaign_ID instead?

File: Gateway/MinistryPlatform.Translation/Repositories/CampaignRepository.cs [hunk 2]
```diff
@@ -53,6 +57,13 @@ public MpPledgeCampaign GetPledgeCampaign(int campaignId)
            
         }
 
-        
+        public List<MpTripRecord> GetGoTripDetailsByCampaign(int pledgeCampaignId)
+        {
+            var apiToken = _apiUserRepository.GetToken();
+            var parms = new Dictionary<string, object> { { "@Pledge_Campaign_ID", pledgeCampaignId } };
+            var tripRecords = _ministryPlatformRest.UsingAuthenticationToken(apiToken).GetFromStoredProc<MpTripRecord>(_configurationWrapper.GetConfigValue("TripRecordProc"), parms);
+            List<MpTripRecord> tripRecord = tripRecords.FirstOrDefault()?? new List<MpTripRecord>();
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation/Repositories/CampaignRepository.cs]: Do you need to use the ?? operator when using first or default?

File: Gateway/crds-angular/Services/TripService.cs [hunk 3]
```diff
@@ -389,7 +426,7 @@ public List<int> SaveParticipants(SaveTripParticipantsDto dto)
                 }
                 CreatePledge(dto, applicant);
                 EventRegistration(events, applicant, dto.Campaign.DestinationId);
-                SendTripParticipantSuccess(applicant.ContactId, events);
+   //             SendTripParticipantSuccess(applicant.ContactId, events);
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/TripService.cs]: Commented code.

File: crossroads.net/app/trips/signup/tripsSignup.controller.js [hunk 4]
```diff
@@ -150,16 +149,17 @@ var attributeTypes = require('crds-constants').ATTRIBUTE_TYPE_IDS;
...
-      switch (vm.signupService.page4.lottery) {
-        case null:
-          return false;
-        case 'As long as I am selected, I will go on the trip.':
-          return false;
-        default:
-          return true;
-      }
-    }
+    //TODO: this seems unessecary now
```
Comment (andycanterbury) [inline crossroads.net/app/trips/signup/tripsSignup.controller.js]: Commented code

File: Gateway/MinistryPlatform.Translation/Repositories/CampaignRepository.cs (hunk 2 above)
Comment (silbermm) [inline Gateway/MinistryPlatform.Translation/Repositories/CampaignRepository.cs]: yes!

**Pull request merged**